packages/manager/src/queries/vpcs/vpcs.ts

```diff
@@ -0,0 +1,169 @@
+  const queryClient = useQueryClient();
+  return useMutation<VPC, APIError[], UpdateVPCPayload>({
+    mutationFn: (data) => updateVPC(id, data),
+    onSuccess: (VPC) => {
+      queryClient.invalidateQueries(vpcQueries.paginated._def);
+      queryClient.setQueryData<VPC>(vpcQueries.vpc(VPC.id).queryKey, VPC);
+    },
+  });
+};
+
+export const useDeleteVPCMutation = (id: number) => {
+  const queryClient = useQueryClient();
+  return useMutation<{}, APIError[]>({
+    mutationFn: () => deleteVPC(id),
+    onSuccess: () => {
+      queryClient.invalidateQueries(vpcQueries.all);
+      queryClient.invalidateQueries(vpcQueries.paginated._def);
+      queryClient.removeQueries(vpcQueries.vpc(id).queryKey);
+    },
+  });
+};
+
+// Subnet queries
+export const useSubnetsQuery = (
+  vpcId: number,
+  params: Params,
+  filter: Filter,
+  enabled: boolean = true
+) =>
+  useQuery<ResourcePage<Subnet>, APIError[]>({
+    ...vpcQueries.vpc(vpcId)._ctx.subnets._ctx.paginated(params, filter),
+    enabled,
+    keepPreviousData: true,
+  });
+
+export const useCreateSubnetMutation = (vpcId: number) => {
+  const queryClient = useQueryClient();
+  return useMutation<Subnet, APIError[], CreateSubnetPayload>({
+    mutationFn: (data) => createSubnet(vpcId, data),
+    onSuccess: () => {
+      // New subnet created --> refresh the VPC queries (all, paginated, & individual), plus the /subnets VPC query
+      queryClient.invalidateQueries(vpcQueries.all);
+      queryClient.invalidateQueries(vpcQueries.paginated._def);
+      queryClient.invalidateQueries(vpcQueries.vpc(vpcId).queryKey);
+      queryClient.invalidateQueries(vpcQueries.vpc(vpcId)._ctx.subnets);
+    },
+  });
+};
+
+export const useUpdateSubnetMutation = (vpcId: number, subnetId: number) => {
+  const queryClient = useQueryClient();
+  return useMutation<Subnet, APIError[], ModifySubnetPayload>({
+    mutationFn: (data) => modifySubnet(vpcId, subnetId, data),
+    onSuccess: () => {
+      // New subnet created --> refresh the VPC queries (all, paginated, & individual), plus the /subnets VPC query
+      queryClient.invalidateQueries(vpcQueries.all);
+      queryClient.invalidateQueries(vpcQueries.paginated._def);
+      queryClient.invalidateQueries(vpcQueries.vpc(vpcId).queryKey);
+      queryClient.invalidateQueries(vpcQueries.vpc(vpcId)._ctx.subnets);
+    },
+  });
+};
+
+export const useDeleteSubnetMutation = (vpcId: number, subnetId: number) => {
+  const queryClient = useQueryClient();
+  return useMutation<{}, APIError[]>({
+    mutationFn: () => deleteSubnet(vpcId, subnetId),
+    onSuccess: () => {
+      // New subnet created --> refresh the VPC queries (all, paginated, & individual), plus the /subnets VPC query
+      queryClient.invalidateQueries(vpcQueries.all);
+      queryClient.invalidateQueries(vpcQueries.paginated._def);
+      queryClient.invalidateQueries(vpcQueries.vpc(vpcId).queryKey);
+      queryClient.invalidateQueries(vpcQueries.vpc(vpcId)._ctx.subnets);
```

Comment:
Unfortunately typescript doesn't catch these causes, but can we update this to be more explicit? Add `queryKey` to the end. This change might be needed in a few places.

```suggestion
      queryClient.invalidateQueries(
        vpcQueries.vpc(vpcId)._ctx.subnets.queryKey
      );
```